### PR TITLE
TELCODOCS-1404: CatalogSource CR depracation note and DefaultCatSrc CR update

### PR DESCRIPTION
**Pull request by amolnar-gh**
Version(s): 4.14

Issue: https://issues.redhat.com/browse/TELCODOCS-1404

Link to docs preview: https://64291--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-14-release-notes#ocp-4-14-ztp-talm-defaultcatsrc-update

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by pixelsoccupied** [comment]
Not 100% sure about title. Can we have something more like --> 
"must use one specific CatalogSource CR (and to push you in that direction we marked the rest as deprecated)" 
instead of 
"don't use CatalogSourceOperator-specific CR"

File: release_notes/ocp-4-14-release-notes.adoc [hunk 1]
```diff
@@ -572,6 +572,11 @@ As of {product-title} 4.14, `DeploymentConfig` objects are deprecated. `Deployme
 
 Instead, use `Deployment` objects or another alternative to provide declarative updates for pods.
 
+[id="ocp-4-14-ztp-talm-defaultcatsrc-update"]
+=== Operator-specific CatalogSource CRs used in {gitops-shortname} ZTP are deprecated
+
+From {product-title} {product-version}, you must only use the `DefaultCatSrc.yaml` `CatalogSource` CR when updating Operators with {cgu-operator-first}. All other `CatalogSource` CRs are deprecated. For more information about `DefaultCatSrc` CR, see xref:../scalability_and_performance/ztp_far_edge/ztp-talm-updating-managed-policies.adoc#talo-operator-update_ztp-talm[Performing an Operator update].
```
Comment (bburt-rh) [inline release_notes/ocp-4-14-release-notes.adoc:578]:
```suggestion
From {product-title} {product-version}, you must only use the `DefaultCatSrc.yaml` `CatalogSource` CR when updating Operators with {cgu-operator-first}. All other `CatalogSource` CRs are deprecated and are planned to removed in the <deprecation_timeline>. Red{nbsp}Hat will provide bug fixes and support for this feature during the current release lifecycle, but this feature will no longer receive enhancements and will be removed. For more information about `DefaultCatSrc` CR, see xref:../scalability_and_performance/ztp_far_edge/ztp-talm-updating-managed-policies.adoc#talo-operator-update_ztp-talm[Performing an Operator update].
```
Consider following more closely the recommended text from [the SSG's deprecation notice template](https://redhat-documentation.github.io/supplementary-style-guide/#release-notes-doc-texts) to ensure that users fully understand what "deprecation" means in this context. I've suggested one possible revision above.

**Comment by pixelsoccupied** [comment]
lgtm

**Comment by josclark42** [comment]
lgtm

**Comment by jab-rh** [comment]
@amolnar-rh, please squash your commits to 1. Thanks!

**Comment by amolnar-rh** [comment]
@jab-rh Sorry about that. Squashed now! Could you please merge it once the check passes?

**Pull request merged**